File: 2023/day1/star1.rs
fn main() {
    let my_str = include_str!("input.txt").split("\n").collect::<Vec<_>>();
    let mut out = 0;

    for row in my_str {
        let mut temp = Vec::new();
        for char in row.chars() {
            if char.is_ascii_digit() {
                temp.push(char);
            }
        }
        if temp.len() > 0 {
            let mut blub = String::from("");
            blub.push(temp[0]);
            blub.push(temp.last().copied().expect(""));

            out += blub.parse::<i32>().unwrap();
        }
    }
    print!("{out}");
}

File: 2025/day1/star1.py
from input import input

number = 0

rotation = 50
input = input.split("\n")

for row in input:
    way = row[0]
    amount = row[1:]

    if(way == "R"):
        rotation += int(amount)
    elif(way == "L"):
        rotation -= int(amount)

    rotation %= 100
    if(rotation == 0):
        number += 1

print(number)
